src/olympia/versions/tests/test_views.py

```diff
@@ -566,43 +585,134 @@ def setUp(self):
 
 class TestDownloadsLatest(TestDownloadsBase):
     def test_404(self):
-        url = reverse('downloads.latest', args=[123])
+        url = reverse('downloads.latest', kwargs={'addon_id': 123})
         assert self.client.get(url).status_code == 404
 
-    def test_type_none(self):
+    def test_urls(self):
+        assert (
+            reverse(
+                'downloads.latest',
+                kwargs={'addon_id': self.addon.slug},
+            )
+            == '/firefox/downloads/latest/better-gcal-5299/'
+        )
+        assert (
+            reverse(
+                'downloads.latest',
+                kwargs={'addon_id': self.addon.slug, 'type_': 'attachment'},
+            )
+            == '/firefox/downloads/latest/better-gcal-5299/type:attachment/'
```

Comment:
😨 it actually puts `type:attachment/` - that's pretty strange!

Reply:
The whole thing with optional captured arguments inside an optional non-capturing group drove me crazy. At some point I had the opposite behavior, where we'd get `/firefox/downloads/latest/better-gcal-5299/attachment`. That's why I added this test (there are more in `files/tests/test_models.py` - I added this one here for convenience when debugging views.